libr/bin/p/bin_mach0.c

```diff
@@ -740,17 +738,17 @@ static RList* patch_relocs(RBin *b) {
 static void swizzle_io_read(struct MACH0_(obj_t) *obj, RIO *io) {
 	r_return_if_fail (io && io->desc && io->desc->plugin);
 	RIOPlugin *plugin = R_NEW0 (RIOPlugin);
-	if (heapplugin) {
+	if (io->heapplugin) {
```

Comment:
the io api have nothing to do with the macho parser or this swizzling hack. in any case. the heapplugin pointer should be saved into obj, because its macho specific